src/Microsoft.AspNet.OData.Shared/ODataAPIHandler.cs

```diff
@@ -0,0 +1,186 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="ODataAPIHandler.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.OData.Edm;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// The handler class for handling users' create, delete and update methods.
+    /// This is the handler for data modification where there is a CLR type.
+    /// </summary>
+    public abstract class ODataAPIHandler<TStructuralType>: IODataAPIHandler where TStructuralType : class
+    {
+        /// <summary>
+        /// TryCreate method to create a new object.
+        /// </summary>        
+        /// <param name="keyValues">The key-value pair of the object to be created. Optional</param>
+        /// <param name="createdObject">The created object (CLR or Typeless)</param>
+        /// <param name="errorMessage">Any error message in case of an exception</param>
+        /// <returns>The status of the TryCreate method <see cref="ODataAPIResponseStatus"/> </returns>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1021:AvoidOutParameters", MessageId = "2#")]
```

Comment:
How come you need to suppress the warning message here and not in `TryGet` and `TryDelete`?